CMakeLists.txt

```diff
@@ -14,6 +14,11 @@ set_target_properties(ArborX PROPERTIES INTERFACE_COMPILE_FEATURES cxx_std_14)
 # within a subdirectory.
 add_dependencies(ArborX record_hash)
 
+if(Kokkos_ENABLE_HIP)
+  find_package(rocthrust REQUIRED CONFIG)
```

Comment:
Would make it optional, and print a warning otherwise.

Reply:
I added a `ARBORX_ENABLE_ROCTHRUST` option and issue a warning below when the HIP backend is being used and it is not enabled.

Reply:
I think we should try to find it by default and tell the user how to disable rocthrust if it can't be found. For me, it's important to have a CMake option.

Reply:
I definitely agree that we need to find it by default. What I still don't understand is when it's not found, what's the point of us telling user to disable it, why can't we do that ourselves. I.e., how do you expect a user to use the CMake option? I.e.,
- If a user disables `rocThrust`, what's the point of running HIP if we know the performance sucks?
- If a user enables `rocThrust`, what do we do if it's not found?

The only positive of having an option is to get user a hard error if he requests it, but it's not found. But I would say a similar goal can be achieved through a big fat warning.

Reply:
@aprokop What you suggest makes things more complicated in the exported configuration because the `REQUIRED` would not be forwarded
https://cmake.org/cmake/help/latest/module/CMakeFindDependencyMacro.html
This is not a problem I am interested in solving at the moment.  The current version looks like an acceptable compromise to me.

Reply:
I'm OK with this compromise, i.e., searching `rocThrust` by default, and requiring user to disable it explicitly.